metadata-ingestion/src/datahub/emitter/rest_emitter.py

```diff
@@ -453,19 +465,24 @@ def _post_fetch_server_config(self) -> None:
                     DEFAULT_REST_EMITTER_ENDPOINT == RestSinkEndpoint.OPENAPI
                 )
 
+    def test_connection(self) -> None:
+        self.fetch_server_config()
         logger.debug(
             f"Using {'OpenAPI' if self._openapi_ingestion else 'Restli'} for ingestion."
         )
         logger.debug(
             f"{EmitMode.ASYNC_WAIT} {'IS' if self._should_trace(emit_mode=EmitMode.ASYNC_WAIT, warn=False) else 'IS NOT'} supported."
         )
 
-    def test_connection(self) -> None:
-        self.fetch_server_config()
-
     def get_server_config(self) -> dict:
         return self.server_config.raw_config
 
+    def invalidate_config_cache(self) -> None:
```

Comment:
This is expected to be used only in tests, right?

In that case, we typically use the `freezegun` library to customize time in tests. So instead of adding this method, you could try something like this: https://github.com/spulec/freezegun?tab=readme-ov-file#moving-time-to-specify-datetime